### PR TITLE
Fix code smells for Spring modules

### PR DESCRIPTION
**Pull request by moarychan**
# Description

- Fixed the changelog format for 6.0.0-beta.3.
- Removed a changelog item as it can be ignored for developers, it's used for class `sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/factory/AbstractAzureServiceClientBuilderFactory.java`.
- Deleted the *spring-reference.yml*, only need to maintain the file in the `main` branch.
- Removed the unused properties in database starters.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by saragluna — APPROVED** [review]

File: sdk/spring/spring-reference.yml [hunk 1]
```diff
@@ -1,847 +0,0 @@
-# Please refer to this link to see design details and how to change:
```
Comment (chenrujun) [inline sdk/spring/spring-reference.yml:1]: How about not change this file, just sync from main branch. Otherwise, we need resolve the conflicts.

**Review by chenrujun — CHANGES_REQUESTED** [review]
Could you please sync to main branch about these diffs? (Except spring-reference file if you still decide to delete this file in this branch.)

**Comment by moarychan** [comment]
> Could you please sync to main branch about these diffs? (Except spring-reference file if you still decide to delete this file in this branch.)

We should discuss the upgrade of these Security deprecations on Spring Boot 2.7.x.

File: sdk/spring/CHANGELOG.md [hunk 2]
```diff
@@ -31,11 +21,20 @@ This section includes changes in `spring-cloud-azure-autoconfigure` module.
...
-- Use the `com.azure.core.util.ConfigurationBuilder#ConfigurationBuilder()` instead the default constructor method `com.azure.core.util.Configuration#Configuration()`[#31543](https://github.com/Azure/azure-sdk-for-java/pull/31543).
 - Use an abstract configurer `AbstractHttpConfigurer` instead of `org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter`[#31543](https://github.com/Azure/azure-sdk-for-java/pull/31543).
 - Use the annotation `org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity` instead the `org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity`[#31808](https://github.com/Azure/azure-sdk-for-java/pull/31808).
 - Use the class `org.springframework.security.oauth2.server.resource.authentication.BearerTokenAuthenticationToken` instead of `org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken`[#31808](https://github.com/Azure/azure-sdk-for-java/pull/31808).
 
+#### Bugs Fixed
```
[the remark this answers is not in the document]
Comment (moarychan) [reply sdk/spring/CHANGELOG.md:28]: OK, I will re-sort the changes for 6.0.0-beta.3

File: sdk/spring/spring-cloud-azure-starter-jdbc-mysql/pom.xml [hunk 3]
```diff
@@ -28,8 +28,6 @@
   </issueManagement>
 
   <properties>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
```
[the remark this answers is not in the document]
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-starter-jdbc-mysql/pom.xml:32]: You may search in the main branch, it's only required for Spring Boot 3 branch.

File: sdk/spring/spring-reference.yml (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-reference.yml:1]: Currently, we don't need to sync from main to Spring boot 3 branch,  the sync job should be decided by the PR target. I prefer to delete this one, the conflicts will be only one time when Spring Boot 3 branch is merged back to the main branch.

**Comment by moarychan** [comment]
> > Could you please sync to main branch about these diffs? (Except spring-reference file if you still decide to delete this file in this branch.)
> 
> We should discuss the upgrade of these Security deprecations on Spring Boot 2.7.x.

Opened an issue to track: https://github.com/Azure/azure-sdk-for-java/issues/32279

File: sdk/spring/CHANGELOG.md [hunk 4]
```diff
@@ -31,44 +21,51 @@ This section includes changes in `spring-cloud-azure-autoconfigure` module.
 - Delete the auto configuration for Spring Cloud Sleuth support[#31543](https://github.com/Azure/azure-sdk-for-java/pull/31543).
 - Use the `com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier#DefaultJWTClaimsVerifier(com.nimbusds.jwt.JWTClaimsSet, java.util.Set<java.lang.String>)` instead of `com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier#DefaultJWTClaimsVerifier()`[#31543](https://github.com/Azure/azure-sdk-for-java/pull/31543).
 - Use the `org.springframework.http.ResponseEntity#getStatusCode` instead of `org.springframework.http.ResponseEntity#getStatusCodeValue`[#31543](https://github.com/Azure/azure-sdk-for-java/pull/31543).
-- Use the `com.azure.core.util.ConfigurationBuilder#ConfigurationBuilder()` instead the default constructor method `com.azure.core.util.Configuration#Configuration()`[#31543](https://github.com/Azure/azure-sdk-for-java/pull/31543).
```
Comment (chenrujun) [inline sdk/spring/CHANGELOG.md:34]: Why delete this one? It's better to add description in the PR description.

Comment (moarychan) [reply sdk/spring/CHANGELOG.md:34]: This change can be ignored by developers, it's used for `sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/factory/AbstractAzureServiceClientBuilderFactory.java`.

**Review by chenrujun — APPROVED** [review]

**Pull request merged**